packages/components/src/box-control/index.tsx

```diff
@@ -189,12 +186,24 @@ function BoxControl( {
 				</LinkedButtonWrapper>
 			) }
 
-			{ ! isLinked && splitOnAxis && (
-				<AxialInputControls { ...inputControlProps } />
-			) }
-			{ ! isLinked && ! splitOnAxis && (
-				<InputControls { ...inputControlProps } />
-			) }
+			{ ! isLinked &&
+				splitOnAxis &&
+				[ 'horizontal', 'vertical' ].map( ( axis ) => (
+					<InputControl
+						key={ axis }
+						side={ axis as 'horizontal' | 'vertical' }
```

Comment:
Could be a good idea to type this better:

https://github.com/WordPress/gutenberg/commit/0791e93c5bdcd9739229798e9a0fa7b7c9314226

We could move `horizontal | vertical` to its own type and reuse it here and for the `sides` prop.